core/src/main/java/io/jenkins/pluginhealth/scoring/scores/AdoptionScoring.java

```diff
@@ -43,31 +45,157 @@ public class AdoptionScoring extends Scoring {
+                .parse(lastCommitDateMessage, DateTimeFormatter.ISO_DATE_TIME)
+                .withZoneSameInstant(getZone());
+            return Duration.between(then, commitDate).abs();
         }
 
-        final ProbeResult lastCommitProbeResult = plugin.getDetails().get(LastCommitDateProbe.KEY);
-        if (lastCommitProbeResult != null && lastCommitProbeResult.status().equals(ResultStatus.SUCCESS)) {
-            final String message = lastCommitProbeResult.message();
-            final ZonedDateTime commitDateTime = ZonedDateTime.parse(message);
-
-            final Duration between = Duration.between(plugin.getReleaseTimestamp().toInstant(), commitDateTime.toInstant());
-            if (between.toDays() <= Duration.of(6 * 30, ChronoUnit.DAYS).toDays()) { // Less than 6 months
-                return new ScoreResult(KEY, 1, COEFFICIENT);
-            } else if (between.toDays() < Duration.of(365, ChronoUnit.DAYS).toDays()) { // Less than a year
-                return new ScoreResult(KEY, .75f, COEFFICIENT);
-            } else if (between.toDays() < Duration.of(2 * 365, ChronoUnit.DAYS).toDays()) { // Less than 2 years
-                return new ScoreResult(KEY, .5f, COEFFICIENT);
-            } else if (between.toDays() < Duration.of(4 * 365, ChronoUnit.DAYS).toDays()) { // Less than 4 years
-                return new ScoreResult(KEY, .25f, COEFFICIENT);
-            }
+        protected ZoneId getZone() {
+            return ZoneId.of("UTC");
+        }
+
+        @Override
+        public int getWeight() {
+            return 1;
         }
+    }
+
+    @Override
+    public List<ScoringComponent> getComponents() {
+        return List.of(
+            new ScoringComponent() {
+                @Override
+                public String getDescription() {
+                    return "The plugin must not be marked as up for adoption.";
+                }
+
+                @Override
+                public ScoringComponentResult getScore(Plugin $, Map<String, ProbeResult> probeResults) {
+                    final ProbeResult probeResult = probeResults.get(UpForAdoptionProbe.KEY);
+                    if (probeResult == null || ProbeResult.Status.ERROR.equals(probeResult.status())) {
+                        return new ScoringComponentResult(-1000, 1000, List.of("Cannot determine if the plugin is up for adoption."));
+                    }
+
+                    return switch (probeResult.message()) {
+                        case "This plugin is not up for adoption." ->
+                            new ScoringComponentResult(100, getWeight(), List.of("The plugin is not marked as up for adoption"));
+                        case "This plugin is up for adoption." ->
+                            new ScoringComponentResult(-1000, getWeight(), List.of("The plugin is marked as up for adoption"));
+                        default -> new ScoringComponentResult(-100, getWeight(), List.of());
+                    };
+                }
```

Comment:
I have issues to see when having two sources of information for the scoring components is good when everything is already in the probe results to be honest. That might requires more discussion if possible.

Reply:
As we discuss yesterday, I think there is some confusion. 
What you call the context, to be passed between the probe execution and the scoring process is really all the probe result, which is the `Plugin#details` field.
It would be great to have the "value" of the `ProbeResult` specific to the probe itself, not always using `String` but potentially `int` or `boolean`. However, this is a limitation I found due to the persistence of the record into the DB. Well technically not the persistence but the deserialization of it when we load the results into POJO.

As the end result of the discussion, it seemed that the current design is making a bit of sense and that it's ok, at least for now. Correct? 
If so, I'll merge and release the project with this pull request.